patches/server/0807-Discard-friction-API.patch

```diff
@@ -0,0 +1,72 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Noah van der Aa <ndvdaa@gmail.com>
+Date: Wed, 15 Sep 2021 20:44:22 +0200
+Subject: [PATCH] Discard friction API
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+index 063f3e4c67e6716c9a03dbe4b72eafd32e4f0d53..f3f6b635a056d3014ca6c78e145e12b6e133bb42 100644
+--- a/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
++++ b/src/main/java/net/minecraft/world/entity/item/ItemEntity.java
+@@ -53,6 +53,7 @@ public class ItemEntity extends Entity {
+     public final float bobOffs;
+     private int lastTick = MinecraftServer.currentTick - 1; // CraftBukkit
+     public boolean canMobPickup = true; // Paper
++    public boolean discardFriction = false; // Paper
+ 
+     public ItemEntity(EntityType<? extends ItemEntity> type, Level world) {
+         super(type, world);
+@@ -139,7 +140,11 @@ public class ItemEntity extends Entity {
+                 this.move(MoverType.SELF, this.getDeltaMovement());
+                 float f1 = 0.98F;
+ 
+-                if (this.onGround) {
++                // Paper start
++                if (this.discardFriction) {
++                    f1 = 1F;
++                } else if (this.onGround) {
++                    // Paper end
+                     f1 = this.level.getBlockState(new BlockPos(this.getX(), this.getY() - 1.0D, this.getZ())).getBlock().getFriction() * 0.98F;
+                 }
+ 
+diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+index 342345eb04d00efb58392ccf209e3c51c1064173..410687a651b85bf3931621581d79789bf0ec7735 100644
+--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
++++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftItem.java
+@@ -85,6 +85,16 @@ public class CraftItem extends CraftEntity implements Item {
+      public void setWillAge(boolean willAge) {
+         item.age = willAge ? 0 : NO_AGE_TIME;
+      }
++
++     @Override
++     public boolean discardsFriction() {
```

Comment:
This isnt a requested change, but does either of `willDiscardFriction` or `isDiscardingFriction` sound better?